store/postgres/src/fork.rs

```diff
@@ -94,11 +94,19 @@ impl SubgraphFork {
         schema: Arc<Schema>,
         logger: Logger,
     ) -> Result<Self, StoreError> {
+        let domain = match base.domain() {
+            Some(domain) => domain,
+            None => return Err(StoreError::ForkFailure(format!("invalid base URL: {}", base))),
+        };
+
         Ok(Self {
             client: reqwest::Client::new(),
-            endpoint: base
-                .join(id.as_str())
-                .map_err(|e| StoreError::ForkFailure(format!("failed to join fork base: {}", e)))?,
+            endpoint: Url::parse(
+                format!("https://{}{}/{}", domain, base.path(), id).as_str(),
```

Comment:
I don't think we should go this path (pun intended), I think we should error when someone uses a URL that doesn't end with `/`.

And if ideally this error should happen on `graph-node` startup, so that the error gets caught early.

I would even add this to the `docs` PR.

What do you think?

Reply:
Related to the discussion: https://github.com/graphprotocol/graph-node/pull/2995#discussion_r795717925

If you agree to early return the error, a simple implementation that I can think of is to assert that the `path` is a `folder`.

Reply:
`FORK_BASE_URL` env? It is not intended to set the fork base using this environment variable. Although I might be wrong. Maybe some rust macro I used to define the `--fork_base` arg does this behind my back?

Anyway, now that I tested it - yes, it has this issue. Huh, I remember testing this scenario...

But imo, missing a `/` at the end should not cause an error. Imo, this is an expected input. In a web browser when you dial a URL and miss the `/` it doesn't complain that the URL is invalid or smth. It seems an intuitive behavior for me.

Reply:
Sorry the env doesn't exist, I forgot I'm running `graph-node` like this 😅 

```
export FORK_BASE_URL=https://api.thegraph.com/subgraphs/id
cargo run -p graph-node -- \
        --postgres-url $THEGRAPH_STORE_POSTGRES_DIESEL_URL \
        --ethereum-rpc $XDAI_RPC_URL \
        --ipfs 127.0.0.1:5001 \
        --fork-base $FORK_BASE_URL
```

My point about the error is because we know before hand we'll concatenate the base URL with a deployment id. Since we know that for sure, I think `graph-node` should validate it before hand (and panic) so that someone doesn't discover that only later when trying to debug the subgraph.

Reply:
But yeah, I don't think this is a must to finish the PR, we can merge with the old way (preferable IMO) where the `file` is replaced.

The error message can be improved for it later.

And I would recommend add a note about it in the docs (optional as well), so people don't shoot themselves in the foot.

Reply:
Ok, sounds good. I will revert the last commit and update the doc with a note.
Afterwards, I will work on a small PR to address the `/` in the fork base url.